src/core/components/forms/inputFileAvatar/inputFileAvatar.test.tsx

```diff
@@ -55,37 +58,57 @@ describe('<InputFileAvatar /> component', () => {
         expect(screen.getByTestId(IconType.PLUS)).toBeInTheDocument();
     });
 
-    it('displays a preview and calls the onFileSelect callback when a valid file is selected', async () => {
+    it('displays a preview and calls the onChange callback when a valid file is selected', async () => {
         const user = userEvent.setup();
         const label = 'test-label';
         const fileSrc = 'https://chucknorris.com/image.png';
-        const file = new File(['(⌐□_□)'], fileSrc, { type: 'image/png' });
-        const onFileSelect = jest.fn();
+        const file = new File(['(⌐□_□)'], 'image.png', { type: 'image/png' });
+        const onChange = jest.fn();
         createObjectURLMock.mockReturnValue(fileSrc);
 
-        render(createTestComponent({ label, onFileSelect }));
-        await user.upload(screen.getByLabelText(label), file);
-        const previewImg = await screen.findByRole<HTMLImageElement>('img');
+        const { rerender } = render(createTestComponent({ label, onChange }));
+
+        const fileInput = screen.getByLabelText<HTMLInputElement>(label);
+        await user.upload(fileInput, file);
+
+        await waitFor(() => {
+            expect(onChange).toHaveBeenCalledWith({ url: fileSrc, file });
+        });
 
+        rerender(createTestComponent({ label, onChange, value: { url: fileSrc, file } }));
+
+        const previewImg = await screen.findByTestId('avatar');
         expect(previewImg).toBeInTheDocument();
-        expect(previewImg.src).toEqual(fileSrc);
-        expect(onFileSelect).toHaveBeenCalledWith(file);
+        expect(previewImg).toHaveAttribute('src', fileSrc);
     });
 
     it('clears the current file selection on close button click after an image has been selected', async () => {
         const user = userEvent.setup();
         const label = 'test-label';
         const file = new File(['something'], 'test.png', { type: 'image/png' });
-        createObjectURLMock.mockReturnValue('file-src');
+        const fileSrc = 'file-src';
+        const onChange = jest.fn();
+        createObjectURLMock.mockReturnValue(fileSrc);
+
+        const { rerender } = render(createTestComponent({ label, onChange, value: { url: fileSrc, file } }));
+
+        const previewImg = await screen.findByTestId('avatar');
+        expect(previewImg).toBeInTheDocument();
 
-        render(createTestComponent({ label }));
-        await user.upload(screen.getByLabelText(label), file);
         const cancelButton = await screen.findByRole('button');
         expect(cancelButton).toBeInTheDocument();
 
         await user.click(cancelButton);
+
+        await waitFor(() => {
+            expect(onChange).toHaveBeenCalled();
```

Comment:
I'd specify that the `onChange` is called with undefined when clearing the input
```suggestion
            expect(onChange).toHaveBeenCalledWith(undefined);
```

Reply:
I added this, but also needed to explicitly set it to undefined in the handleCancel on the component itself